### PR TITLE
Rework `docker-testbed` workflow

### PR DESCRIPTION
**Pull request by FirelightFlagboy**
- Use a new format for the image tag:

  The new format is `{{ version }}-{{ YYYYMMDD }}.sha.{{ short-sha }}`.
  You have a step that summary the pushed image.

- Rename the workflow from `publish-testbed` to `docker-testbed`

  To harmonize with the yet to exist workflow to build the docker image
  for the parsec backend.

- Prevent concurrency run.

- Use the gihtub action `docker/setup-buildx-action` to setup build-kit
  engine for docker, that provide support for the cache system & multi
  platforms builds.

- Use the GitHub action `docker/metadata-actions`:

  - Generate the GitHub tags for us.
  - Handle the image name (would could setup to push to `ghcr.io` &
    `docker.io` in the future).

- Use the GitHub action `docker/build-publish-actions`:

  - Simplify configuring the docker build phase.
  - Leverage the GitHub actions cache.
  - Would allow to build for multiple platforms.
  - Allow to simplify push the generate image with its multiple tags.

- Rework the test part, mainly split into 3 steps instead of 1.

### CODE REVIEW

File: .github/workflows/docker-testbed.yml [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+              print('Success')
+              break
+          else:
+            raise SystemExit('Cannot connect to testbed server :(')
+        shell: python
+
+      - name: Stop docker test container
+        run: docker container stop ${{ steps.test-container.outputs.id }}
+
+      - name: Image that will be push
```
Comment (mmmarcos) [inline .github/workflows/docker-testbed.yml:118]:
```suggestion
      - name: Image to be published
```

File: .github/workflows/docker-testbed.yml [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+              urlopen(r)
+            except Exception as exc:
+              print(f'Try {i + 1}/10: {exc}')
+              time.sleep(1)
+              continue
+            else:
+              print('Success')
+              break
+          else:
+            raise SystemExit('Cannot connect to testbed server :(')
```
Comment (mmmarcos) [inline .github/workflows/docker-testbed.yml:112]: Can we move this script to a file next to the .yml? I know it only tests the docker image but we may want to improve/update it independently of the workflow.

**Review by mmmarcos — APPROVED** [review]
LGTM, just a couple of comments

File: .github/workflows/docker-testbed.yml (hunk 2 above)
Comment (FirelightFlagboy) [reply .github/workflows/docker-testbed.yml]: done

File: .github/scripts/test-testbed-server.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            print(f"Try {i + 1}/{MAX_RETRIES}: {exc}")
+            time.sleep(DURATION_BETWEEN_RETRIES)
+            continue
+        else:
+            return True
+    return False
+
+
+if __name__ == "__main__":
+    r = Request("http://127.0.0.1:6777/testbed/new/empty", method="POST")
```
Comment (mmmarcos) [inline .github/scripts/test-testbed-server.py:24]: If this is not printed somewhere else, maybe is a good idea to have it in the logs?
```suggestion
    r = Request("http://127.0.0.1:6777/testbed/new/empty", method="POST")
    print(f"Will connect to testbed server on: {r.get_full_url()}")
```

**Review by mmmarcos — APPROVED** [review]

**Pull request merged**